report/node.go

```diff
@@ -80,10 +79,26 @@ func (n Node) WithLatest(k string, ts time.Time, v string) Node {
 	return n
 }
 
-// WithCounters returns a fresh copy of n, with Counters c merged in.
-func (n Node) WithCounters(c map[string]int) Node {
-	n.Counters = n.Counters.Merge(Counters{}.fromIntermediate(c))
-	return n
+// LookupCounter returns the value of a counter
+// (counters are stored as strings, to keep the data structure simple)
+func (n Node) LookupCounter(k string) (value int, found bool) {
+	name := CounterPrefix + k
+	var str string
+	if str, found = n.Latest.Lookup(name); found {
+		value, _ = strconv.Atoi(str)
+	}
+	return value, found
+}
+
+// WithCounter returns a fresh copy of n, with Counter c added in.
+// (counters are stored as strings, to keep the data structure simple)
+func (n Node) WithCounter(k string, value int) Node {
```

Comment:
```suggestion
func (n Node) AddCounter(k string, value int) Node {
```

I suggest this change for two reasons:

1. `AddCounter` suggests more explicitly that if the new value will be _added_ to the old counter value if it existed, rather than overwritting it
2. `Counters.Add` -> `AddCounter` is a smoother change, also consistent with `Counters.Lookup` -> `LookupCounter`

WDYT?